### PR TITLE
Origin/backend/feature/portfolio api

### PR DESCRIPTION
**Pull request by NazlicanAka**
* Portfolio service, controller, DTO's are implemented.
* Portfolio repository is updated.

### CODE REVIEW

File: backend/src/main/resources/application.properties
[the remark this answers is not in the document]
Comment (NazlicanAka) [reply backend/src/main/resources/application.properties:1]: Okey, I have changed the file as it was. Does this make the commit okey, or I need to remove the file from commit completely.

File: backend/src/main/java/com/bounswe2024group10/Tradeverse/service/PortfolioService.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    public GetAllPortfoliosResponse getAllPortfolios(GetAllPortfoliosRequest request) {
+        List<Portfolio> portfolios = portfolioRepository.findByUsername(request.getUsername());
+
+        if (portfolios.isEmpty()) {
+            return new GetAllPortfoliosResponse(false, "No portfolios found for this user", null);
+        }
+
+        List<PortfolioDto> portfolioDtos = portfolios.stream()
+                .map(portfolio -> new PortfolioDto(portfolio.getId(), portfolio.getName(), portfolio.getVisibility(), portfolio.getCreatedAt()))
```
Comment (melihakpinar) [inline backend/src/main/java/com/bounswe2024group10/Tradeverse/service/PortfolioService.java:37]: What is createdAt and why are we storing it?

It is not defined at PortfolioDto class.

File: backend/src/main/java/com/bounswe2024group10/Tradeverse/controller/PortfolioController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.bounswe2024group10.Tradeverse.controller;
+
+import com.bounswe2024group10.Tradeverse.dto.portfolio.*;
+import com.bounswe2024group10.Tradeverse.service.PortfolioService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/portfolio")
```
Comment (melihakpinar) [inline backend/src/main/java/com/bounswe2024group10/Tradeverse/controller/PortfolioController.java:10]: Add these endpoints to JwtUtil

File: backend/src/main/java/com/bounswe2024group10/Tradeverse/dto/portfolio/CreatePortfolioRequest.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package com.bounswe2024group10.Tradeverse.dto.portfolio;
+
+public class CreatePortfolioRequest {
```
Comment (melihakpinar) [inline backend/src/main/java/com/bounswe2024group10/Tradeverse/dto/portfolio/CreatePortfolioRequest.java:3]: Portfolio visibility is not an attribute of portfolio, we are storing it in User class so you can remove it

File: backend/src/main/java/com/bounswe2024group10/Tradeverse/dto/portfolio/CreatePortfolioRequest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    private String username;
+    private String name;
+    private String visibility;
```
Comment (melihakpinar) [inline backend/src/main/java/com/bounswe2024group10/Tradeverse/dto/portfolio/CreatePortfolioRequest.java:6]: Where is the amount variable?

**Review by melihakpinar — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/bounswe2024group10/Tradeverse/dto/portfolio/CreatePortfolioRequest.java (hunk 3 above)
Comment (NazlicanAka) [reply backend/src/main/java/com/bounswe2024group10/Tradeverse/dto/portfolio/CreatePortfolioRequest.java:3]: removed.

File: backend/src/main/java/com/bounswe2024group10/Tradeverse/dto/portfolio/CreatePortfolioRequest.java (hunk 4 above)
Comment (NazlicanAka) [reply backend/src/main/java/com/bounswe2024group10/Tradeverse/dto/portfolio/CreatePortfolioRequest.java:6]: Added as BigDecimal, but I am not sure whether using BigDecimal is okey.

File: backend/src/main/java/com/bounswe2024group10/Tradeverse/service/PortfolioService.java (hunk 1 above)
Comment (NazlicanAka) [reply backend/src/main/java/com/bounswe2024group10/Tradeverse/service/PortfolioService.java:37]: CreatedAt is removed.

File: backend/src/main/java/com/bounswe2024group10/Tradeverse/controller/PortfolioController.java (hunk 2 above)
Comment (NazlicanAka) [reply backend/src/main/java/com/bounswe2024group10/Tradeverse/controller/PortfolioController.java:10]: Okey, added.

**Pull request merged**